mlir/docs/PatternRewriter.md

```diff
@@ -320,15 +320,33 @@ conversion target, via a set of pattern-based operation rewriting patterns. This
 framework also provides support for type conversions. More information on this
 driver can be found [here](DialectConversion.md).
 
+### Walk Pattern Rewrite Driver
+
+This is a fast and simple driver that walks the given op and applies patterns
+that locally have the most benefit. The benefit of a pattern is decided solely
+by the benefit specified on the pattern, and the relative order of the pattern
+within the pattern list (when two patterns have the same local benefit).
+
+This driver does not (re)visit modified or newly replaced ops, and does not
+allow for progressive rewrites of the same op. Op erasure is only supported for
```

Comment:
The basic requirement for any erasure is that it shouldn't invalidate the next pointer saved by `make_early_inc_range`. Making the verification a bit stricter for ops seemed reasonable to me, but I'm not quite sure how this should translate to blocks. Do you have any suggestions?

Reply:
When you matched an op `op`, I would allow erasing blocks that are nested inside of `op`, but not any other blocks. I think erasing a parent block of `op` breaks the walk.